jbi/models.py

```diff
@@ -371,19 +380,17 @@ def map_as_comments(
 
         return [json.dumps(comment, indent=4) for comment in comments]
 
-    def getbug_as_bugzilla_object(self) -> BugzillaBug:
-        """Helper method to get up to date bug data from Request.bug.id in BugzillaBug format"""
-        current_bug_info = get_bugzilla().getbug(self.bug.id)  # type: ignore
-        return BugzillaBug.parse_obj(current_bug_info.__dict__)
-
     @functools.cached_property
     def bugzilla_object(self) -> BugzillaBug:
         """Returns the bugzilla bug object, querying the API as needed for private bugs"""
+        from jbi.services import get_bugzilla  # pylint: disable=import-outside-toplevel
```

Comment:
This seems like more of a "structural" problem that we could address.

What if instead of instead of `bugzilla_object` and `map_as_jira_comment` being object object methods, they were functions in the `services` module instead?

Reply:
I agree.

And I think I want to remove the `bugzilla_object` property, since it's redundant with the `bug` attribute. Especially if we land #219.

Instead, I would propose something like this in `runner.py`:

```py
        try:
            request = request.copy(update={"bug": bugzilla_fetch_private(request.bug)})
        except Exception as ex:
            logger.exception("Failed to get bug: %s", ex, extra=log_context)
            raise IgnoreInvalidRequestError(
                "bug not accessible or bugzilla down"
            ) from ex
```

where `bugzilla_fetch_private()` would be:

```py
def bugzilla_fetch_private(bug: BugzillaBug) -> BugzillaBug:
    """Fetch bug private information from API"""
    client = get_bugzilla()
    # If bug is private, fetch all attributes.
    if bug.is_private:
        bug = client.get_bug(bug.id)
    # If comment is private, then webhook does not have comment, fetch it from server
    if bug.comment.is_private:
        comment_list = client.get_comments(bug.id)
        matching_comments = [c for c in comment_list if c.id == bug.comment.id]
        if len(matching_comments) != 1:
            bug.comment = None
        bug.comment = matching_comments[0]
    return bug
```

This way, we have `request.bug` with the proper bug object, and `request.bug.comment` with the proper comment object.

What do you think?

Reply:
I like it!

Reply:
I will do it in https://github.com/mozilla/jira-bugzilla-integration/pull/220